stdlib/bytes.ml

```diff
@@ -67,7 +67,10 @@ let sub s ofs len =
 let sub_string b ofs len = unsafe_to_string (sub b ofs len)
 
 let extend s left right =
-  let len = length s + left + right in
+  let len = length s + left in
+  if len < 0 then invalid_arg "Bytes.extend" else
+  let len = len + right in
+  if len < 0 then invalid_arg "Bytes.extend" else
```

Comment:
I find the shadowing used here not very readable. You could name the first variable `len_left`, or declare `let (++) a b = (* check for overflow *) let c = a + b in if c < 0 then invalid_arg "Bytes.extend" else c` and `let len = length s ++ left ++ right`.